tests/codes/eurocode/nen_en_1993_5_2008/chapter_5_ultimate_limit_states/test_formula_5_9.py

```diff
@@ -0,0 +1,96 @@
+
+
+class TestForm5Dot9ReducedBendingMomentResistance:
+    """Validation for formula 5.9 from NEN-EN 1993-5:2008."""
+
+    @pytest.mark.parametrize(
+        ("beta_b", "w_pl", "rho", "a_v", "t_w", "alpha", "f_y", "gamma_m_0", "mc_rd", "expected_value"),
+        [
+            (0.8, 500, 0.25, 100, 10, 45, 250, 1.0, 400, 99.9779),  # Mv,Rd < Mc,Rd
+            (0.8, 500, 0.25, 100, 10, 45, 250, 1.0, 50, 50),  # Mv,Rd < Mc,Rd
+        ],
+    )
+    def test_evaluation(  # noqa: PLR0913
+        self,
+        beta_b: float,
+        w_pl: float,
+        rho: float,
+        a_v: float,
+        t_w: float,
+        alpha: float,
+        f_y: float,
+        gamma_m_0: float,
+        mc_rd: float,
+        expected_value: float,
+    ) -> None:
+        """Test the evaluation of the result."""
+        form = Form5Dot9ReducedBendingMomentResistance(
+            beta_b=beta_b, w_pl=w_pl, rho=rho, a_v=a_v, t_w=t_w, alpha=alpha, f_y=f_y, gamma_m_0=gamma_m_0, mc_rd=mc_rd
+        )
+
+        assert form == pytest.approx(expected_value)
+
+    @pytest.mark.parametrize(
+        ("beta_b", "w_pl", "rho", "a_v", "t_w", "alpha", "f_y", "gamma_m_0", "mc_rd"),
+        [
+            (-0.8, 500, 0.25, 100, 10, 45, 250, 1.0, 400),  # beta_b is negative
+            (0.8, -500, 0.25, 100, 10, 45, 250, 1.0, 400),  # w_pl is negative
+            (0.8, 500, -0.25, 100, 10, 45, 250, 1.0, 400),  # rho is negative
+            (0.8, 500, 0.25, -100, 10, 45, 250, 1.0, 400),  # a_v is negative
+            (0.8, 500, 0.25, 100, -10, 45, 250, 1.0, 400),  # t_w is negative
+            (0.8, 500, 0.25, 100, 10, -45, 250, 1.0, 400),  # alpha is negative
+            (0.8, 500, 0.25, 100, 10, 45, 250, -1.0, 400),  # gamma_m_0 is negative
+            (0.8, 500, 0.25, 100, 10, 45, 250, 1.0, -400),  # mc_rd is negative
+            (0.0, 500, 0.25, 100, 10, 45, 250, 1.0, 400),  # beta_b is zero
+            (0.8, 0.0, 0.25, 100, 10, 45, 250, 1.0, 400),  # w_pl is zero
+            (0.8, 500, 0.0, 100, 10, 45, 250, 1.0, 400),  # rho is zero
+            (0.8, 500, 0.25, 0.0, 10, 45, 250, 1.0, 400),  # a_v is zero
+            (0.8, 500, 0.25, 100, 0.0, 45, 250, 1.0, 400),  # t_w is zero
+            (0.8, 500, 0.25, 100, 10, 0.0, 250, 1.0, 400),  # alpha is zero
+            (0.8, 500, 0.25, 100, 10, 45, 250, 0.0, 400),  # gamma_m_0 is zero
+            (0.8, 500, 0.25, 100, 10, 45, 250, 1.0, 0.0),  # mc_rd is zero
+        ],
+    )
+    def test_raise_error_when_invalid_values_are_given(  # noqa: PLR0913
+        self, beta_b: float, w_pl: float, rho: float, a_v: float, t_w: float, alpha: float, f_y: float, gamma_m_0: float, mc_rd: float
+    ) -> None:
+        """Test a zero and negative value for parameters."""
+        with pytest.raises(LessOrEqualToZeroError):
+            Form5Dot9ReducedBendingMomentResistance(
+                beta_b=beta_b, w_pl=w_pl, rho=rho, a_v=a_v, t_w=t_w, alpha=alpha, f_y=f_y, gamma_m_0=gamma_m_0, mc_rd=mc_rd
+            )
```

Comment:
This is very unreadable, there must be a better way. What about this:

```suggestion
    @pytest.mark.parametrize(
        ("invalid_argument", "invalid_value"),
        list(itertools.product(("beta_b", "w_pl", "rho", "a_v", "t_w", "alpha", "f_y", "gamma_m_0", "mc_rd"), (-1.0, 0.0))),
    )
    def test_raise_error_when_invalid_values_are_given(  # noqa: PLR0913
        self, invalid_argument, invalid_value
    ) -> None:
        """Test a zero and negative value for parameters."""

        kwargs = {
            key: (invalid_value if key == invalid_argument else 1.0)
            for key in ("beta_b", "w_pl", "rho", "a_v", "t_w", "alpha", "f_y", "gamma_m_0", "mc_rd")
        }

        with pytest.raises(LessOrEqualToZeroError):
            Form5Dot9ReducedBendingMomentResistance(**kwargs)
```

We are not really interested in the values, just that a negative or zero value raises an error for a specific argument.

Reply:
Nice and clean solution 👍